packet/bgp.go

```diff
@@ -141,10 +141,18 @@ func (b *bgp) enableBGP() error {
 	// first check if it is enabled before trying to create it
 	bgpConfig, _, err := b.client.BGPConfig.Get(b.project, &packngo.GetOptions{})
 	// if we already have a config, just return
-	if err == nil || bgpConfig != nil {
+	// we need some extra handling logic because the API always returns 200, even if
+	// not BGP config is in place.
+	// We treat it as valid config already exists only if ALL of the above is true:
+	// - no error
+	// - bgpConfig struct exists
+	// - bgpConfig struct has non-blank ID
+	// - bgpConfig struct does not have Status=="disabled"
+	if err == nil && bgpConfig != nil && bgpConfig.ID != "" && strings.ToLower(bgpConfig.Status) != "disabled" {
```

Comment:
Noting that the possible values for `bgpConfig.Status` are enabled, disabled, and requested.
This set of comparisons looks good.

Reply:
We may want to narrow down the range of errors we are ok with in the future (404s only?).

Reply:
Agreed. We likely will have to redo this anyways once the BGP config endpoints are better structured.